packages/pulumi-aws/src/apps/storage/StorageApp.ts

```diff
@@ -0,0 +1,77 @@
+import {
+    defineApp,
+    createGenericApplication,
+    ApplicationContext,
+    ApplicationHooks,
+    ApplicationConfig
+} from "@webiny/pulumi-sdk";
+
+import { createCognitoResources } from "./StorageCognito";
+import { createDynamoTable } from "./StorageDynamo";
+import { createFileManagerBucket } from "./StorageFileManager";
+
+export interface StorageAppConfig extends Partial<ApplicationHooks> {
+    protect?(ctx: ApplicationContext): boolean;
+    legacy?(ctx: ApplicationContext): StorageAppLegacyConfig;
+}
+
+export interface StorageAppLegacyConfig {
+    useEmailAsUsername?: boolean;
+}
+
+export const StorageApp = defineApp({
+    name: "storage",
+    config(app, config: StorageAppConfig) {
+        const protect = config.protect?.(app.ctx) ?? app.ctx.env !== "dev";
+        const legacyConfig = config?.legacy?.(app.ctx) ?? {};
+
+        // Setup DynamoDB table
+        const table = createDynamoTable(app, { protect });
+
+        // Setup Cognito
+        const cognito = createCognitoResources(app, {
+            protect,
+            useEmailAsUsername: legacyConfig.useEmailAsUsername ?? false
+        });
+
+        // Setup file storage bucket
+        const fileManagerBucket = createFileManagerBucket(app, { protect });
+
+        app.addOutputs({
+            fileManagerBucketId: fileManagerBucket.output.id,
+            primaryDynamodbTableArn: table.output.arn,
+            primaryDynamodbTableName: table.output.name,
+            cognitoUserPoolId: cognito.userPool.output.id,
+            cognitoUserPoolArn: cognito.userPool.output.arn,
+            cognitoUserPoolPasswordPolicy: cognito.userPool.output.passwordPolicy,
+            cognitoAppClientId: cognito.userPoolClient.output.id
+        });
+
+        return {
+            table,
```

Comment:
Maybe to reduce the amount of changes between legacy and new setups, let's rename the export here to `dynamoDbTable`.

You can check the existing export names in `packages/cwp-template-aws/template/ddb-es/api/pulumi/dev/index.ts:156`. I think being consistent with these should be the best thing to do so that we minimize the possibility of something going wrong upon doing the migration.

Reply:
Sure, just remember that this is not pulumi output - it's a reference to the application resource to be configured by user or used for something else.